streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -889,6 +901,11 @@ public class StreamsConfig extends AbstractConfig {
                     LogAndFailExceptionHandler.class.getName(),
                     Importance.MEDIUM,
                     DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_DOC)
+            .define(DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG,
+                    Type.CLASS,
+                    LogAndFailExceptionHandler.class.getName(),
```

Comment:
@mjsax if we configure a default like this, always new one will be picked up. What do you suggest ?

Reply:
Not sure if I understand the question. Can you elaborate?

Reply:
Currently we have set default values for both old and new config which is LogAndFailExceptionHandler.class.
With this situation, even if user configures a different handler to old config, LogAndFailExceptionHandler will be picked up.

Reply:
Pls ignore this comment. After using the originals().containsKey .. it is fixed.